clap/multiarg.zig

```diff
@@ -0,0 +1,166 @@
+const clap = @import("../clap.zig");
+const std = @import("std");
+
+const testing = std.testing;
+const heap = std.heap;
+const mem = std.mem;
+const debug = std.debug;
+
+pub fn MultiArgComptimeClap(comptime Id: type, comptime params: []const clap.Param(Id)) type {
+    var flags: usize = 0;
+    var single_options: usize = 0;
+    var multi_options: usize = 0;
+    var converted_params: []const clap.Param(usize) = &[_]clap.Param(usize){};
+    for (params) |param| {
+        var index: usize = 0;
+        if (param.names.long != null or param.names.short != null) {
+            const ptr = switch (param.takes_value) {
+                .None => &flags,
+                .One => &single_options,
+                .Many => &multi_options,
+            };
+            index = ptr.*;
+            ptr.* += 1;
+        }
+
+        const converted = clap.Param(usize){
+            .id = index,
+            .names = param.names,
+            .takes_value = param.takes_value,
+        };
+        converted_params = converted_params ++ [_]clap.Param(usize){converted};
+    }
+
+    return struct {
+        single_options: [single_options]?[]const u8,
+        multi_options: [multi_options]std.ArrayList([]const u8),
```

Comment:
Let's have multi_options be `: [multi_options][]const []const u8`. We don't intend to modify this field once we're done parsing.